src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx

```diff
@@ -48,30 +50,36 @@ export const LegacyDiscoverTable = ({
   onAddColumn,
   onFilter,
   onClose,
-  sampleSize,
-  isShortDots,
-  hideTimeColumn,
-  defaultSortOrder,
   showPagination,
   scrollToTop,
 }: DefaultDiscoverTableProps) => {
+  const services = getServices();
+  const [sampleSize, isShortDots, hideTimeColumn, defaultSortOrder] = useMemo(() => {
+    return [
+      services.uiSettings.get(SAMPLE_SIZE_SETTING),
+      services.uiSettings.get(UI_SETTINGS.SHORT_DOTS_ENABLE),
+      services.uiSettings.get(DOC_HIDE_TIME_COLUMN_SETTING),
+      services.uiSettings.get(SORT_DEFAULT_ORDER_SETTING, 'desc') as SortDirection,
+    ];
+  }, [services.uiSettings]);
+
   const displayedColumns = getLegacyDisplayedColumns(
     columns,
     indexPattern,
     hideTimeColumn,
     isShortDots
   );
   const displayedColumnNames = displayedColumns.map((column) => column.name);
-  const pageSize = 50;
-  const [renderedRowCount, setRenderedRowCount] = useState(50); // Start with 50 rows
+  const pageSize = 10;
```

Comment:
Is this an intended change in pageSize from previous behavior?

Reply:
Yes. Being infinite-scrolled, 10 is nice balance between performance and initial view. In a future iteration, we will add lazy loading of the subsequent rows to make it even more unnoticeable.